libbeat/monitoring/metrics.go

```diff
@@ -232,3 +234,57 @@ func fullName(r *Registry, name string) string {
 	}
 	return r.name + "." + name
 }
+
+// Timestamp is a timestamp variable satisfying the Var interface.
+type Timestamp struct {
+	mu     sync.RWMutex
+	ts     time.Time
+	cached string
+}
+
+// NewTimestamp creates and registeres a new timestamp variable.
+func NewTimestamp(r *Registry, name string, opts ...Option) *Timestamp {
+	if r == nil {
+		r = Default
+	}
+
+	v := &Timestamp{}
+	addVar(r, name, opts, v, makeExpvar(func() string {
+		return v.toString()
```

Comment:
We are also missing a lock here. Concurrent operations on Timestamp are racy anyways (which is okay). I think it is enough to just lock in `Set` and `toString`.